### PR TITLE
Use maven-resolver locking instead of takari

### PR DESCRIPTION
**Pull request by wendigo**
No release notes needed

### CODE REVIEW

**Comment by wendigo** [comment]
https://github.com/trinodb/trino/pull/20287

File: .mvn/jvm.config [hunk 1]
```diff
@@ -11,3 +11,5 @@
 --add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED
 -XX:+UnlockDiagnosticVMOptions
 -XX:GCLockerRetryAllocationCount=100
+-Daether.syncContext.named.factory=file-lock
```
Comment (cstamas) [inline .mvn/jvm.config:14]: These should go into `.mvn/maven.config` instead.

Comment (wendigo) [reply .mvn/jvm.config:14]: we set the MAVEN_CONFIG environment variable already in the CI. Will this merge with the .`mvn/maven.config`?

**Comment by cstamas** [comment]
Something is funny: I do have _same error_ on master as this PR:
```
[INFO] Artifact io.confluent:kafka-protobuf-provider:pom:7.5.1 is present in the local repository, but cached from a remote repository ID that is unavailable in current build context, verifying that is downloadable from [sonatype-nexus-snapshots (https://oss.sonatype.org/content/repositories/snapshots, default, snapshots), central (https://repo.maven.apache.org/maven2, default, releases)]
```

Problem is that project seems off:
* io.confluent:kafka-protobuf-provider is ONLY available from confluent
* error tells that only reposes sonatype-nexus-snapshots and central are available in the scope
* so this is build problem, as something not present in 2 available repositories cannot be get from any of those

**Comment by cstamas** [comment]
And I will just leave this here https://issues.apache.org/jira/browse/MDEP-904

**Comment by cstamas** [comment]
Let's do this first (will solve repo scoping) https://github.com/trinodb/trino/pull/20477

**Comment by cstamas** [comment]
Reproduced locally, and it turns out failure was correct. 

@wendigo created another PR to verify it (have CI run it). https://github.com/trinodb/trino/pull/20487

**Pull request closed without merging**